File: app/src/main/java/yoyo/app/android/com/yoyoapp/Trip/booking/BookingViewModel.java
package yoyo.app.android.com.yoyoapp.Trip.booking;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import org.json.JSONObject;

public class BookingViewModel extends AndroidViewModel
{
    private BookingRepository bookingRepository;
    private MutableLiveData<String> bookingIdMutableLiveData;

    public BookingViewModel(@NonNull Application application) {
        super(application);
        bookingRepository = BookingRepository.getInstance(getApplication());
    }


    public void initBookingRequest(String scheduleId, JSONObject jsonObject) {
        bookingIdMutableLiveData = new MutableLiveData<>();
        bookingIdMutableLiveData = bookingRepository.getBookingId(scheduleId,jsonObject);
    }

    public LiveData<String> getBookingId() {
        return bookingIdMutableLiveData;
    }
}

File: app/src/main/java/yoyo/app/android/com/yoyoapp/trip/booking/firstPage/BookingViewModel.kt
package yoyo.app.android.com.yoyoapp.trip.booking.firstPage

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import yoyo.app.android.com.yoyoapp.DataModels.Traveller
import yoyo.app.android.com.yoyoapp.trip.api.BookScheduleRequest

class BookingViewModel(application: Application) : AndroidViewModel(application) {
    private val bookingRepository = BookingRepository(getApplication())
    var paymentUrlMutableLiveData: MutableLiveData<String> = MutableLiveData()
    var travellerCompanionsMutableList: MutableLiveData<List<Traveller>> = MutableLiveData()

    fun initBookingRequest(scheduleId: String, request: BookScheduleRequest) {
        bookingRepository.bookSchedule(scheduleId, request)
        {
            paymentUrlMutableLiveData.value = it.paymentUrl
        }
    }

    fun initGetTravellers() {
        bookingRepository.getCompanionTraveller {
            travellerCompanionsMutableList.value = it
        }
    }


}
